tooling/lsp/src/notifications/mod.rs

```diff
@@ -130,11 +130,13 @@ fn process_noir_document(
 
     let mut workspace_file_manager = file_manager_with_stdlib(&workspace.root_dir);
     insert_all_files_for_workspace_into_file_manager(&workspace, &mut workspace_file_manager);
+    let parsed_files = parse_all(&workspace_file_manager);
```

Comment:
why was the parsing done here instead of inside of the prepare_package method?

Reply:
I chose to do it upstream for two reasons:
 - If parsing is done externally to prepare package (controlled by nargo_cli/lsp), it can be cached as in this example https://github.com/noir-lang/noir/pull/4083
 - But the biggest reason: when working in a workspace, we create a workspace file manager and we run prepare_package for every package in the workspace. If I did parse on prepare_package, I'd be parsing twice any shared file among the workspace members, which is probably most of them. By putting parsing at the same level as the file manager, we have the biggest context of the work to do

Reply:
(If the caller knows that he only needs to check on one specific package of the workspace, he can just parse that one with this approach)

Reply:
In fact if I parsed on prepare_package I'd be parsing all the files in the workspace N times where N is the number of calls to prepare package

Reply:
Since the workspace file manager contains all files for all the crates in the workspace